provider/telegram.go

```diff
@@ -412,14 +430,12 @@ func (tg *tgAPI) request(ctx context.Context, method string, data interface{}) e
 	})
 }
 
-func (tg *tgAPI) parseError(r io.Reader) error {
-	var tgErr = struct {
+func (tg *tgAPI) parseError(r io.Reader, statusCode int) error {
+	tgErr := struct {
 		Description string `json:"description"`
 	}{}
-
-	if err := json.NewDecoder(r).Decode(&tgErr); err != nil {
-		return errors.Wrap(err, "can't decode error")
+	if err := json.NewDecoder(r).Decode(&tgErr); err == nil {
```

Comment:
i liked flipped check more, i.e. err != nil as it seems to be closer the overall code style